### ISSUE TITLE
RangeFacetHandler auto ranges

### ISSUE THREAD

**Issue by dtosato** [zhengchun/Bobo-Browse.Net#18]
Hi all, `RangeFacetHandler` with auto ranges has been removed from the last release on nuget (maybe it should correspond to commit 37a5fa3f9dfa17b3084ac9c1ec135d2457e15f1d). Is that right?

**Comment by NightOwl888** [comment]
The auto range feature does not exist in the Java version of Bobo-Browse 3.2.0, so it also does not exist in the .NET 3.2.0 version. I am not sure why that feature was removed, but my goal is to keep parity with the Java version so I don't plan to add it.

However, if you really need it, it doesn't look like it would be too difficult to inherit RangeFacetHandler, add the constructor overloads that have it, and override the [GetFacetCountCollectorSource](https://github.com/NightOwl888/BoboBrowse.Net/blob/master/src/BoboBrowse.Net/BoboBrowse.Net/Facets/Impl/RangeFacetHandler.cs#L166) method, replacing it with a class that returns a custom IFacetCountCollector that has auto ranges. It looks like the only thing that differs is the [BuildDynamicRanges method](https://github.com/zhengchun/Bobo-Browse.Net/blob/master/src/BoboBrowse.Net/Facets/Impl/RangeFacetCountCollector.cs#L211), which you will need to add.

So, in short you would need 3 custom types:

1. A class that inherits RangeFacetHandler and overrides GetFacetCountCollectorSource + the constructors.
2. A class that inherits RangeFacetCountCollector and overrides GetFacets and GetFacetsNew and adds BuildDynamicRanges  to provide the auto range functionality + override the constructors.
3. A class that is similar to the [RangeFacetHandlerFacetCountCollectorSource](https://github.com/NightOwl888/BoboBrowse.Net/blob/master/src/BoboBrowse.Net/BoboBrowse.Net/Facets/Impl/RangeFacetHandler.cs#L171) that returns the custom RangeFacetCountCollector. This class would then be returned from the GetFacetCountCollectorSource method of the custom RangeFacetHandler.
